rotkehlchen/data_migrations/migrations/migration_3.py

```diff
@@ -6,17 +7,31 @@
 from rotkehlchen.logging import RotkehlchenLogsAdapter
 
 if TYPE_CHECKING:
-    from rotkehlchen.rotkehlchen import Rotkehlchen
+    from rotkehlchen.rotkehlchen import IconManager, Rotkehlchen
 
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 
+def _validate_asset_icons(icon_manager: 'IconManager') -> None:
+    """
+    Loops through icons in the user's data directory and deletes those that are malformed.
+    """
+    icons_directory = icon_manager.icons_dir
+    for icon_entry in icons_directory.iterdir():
+        icon_file_type = imghdr.what(str(icon_entry))
+        if icon_file_type is None:
+            icon_entry.unlink(missing_ok=True)
+
+
 def data_migration_3(rotki: 'Rotkehlchen') -> None:
     """
-    Update the list of ignored assets with tokens from cryptoscamdb
+    - Update the list of ignored assets with tokens from cryptoscamdb.
+    - Delete malformed assets icons.
     """
     try:
         update_spam_assets(db=rotki.data.db)
     except RemoteError as e:
         log.error(f'Failed to update the list of ignored assets during db migration. {str(e)}')
+
+    _validate_asset_icons(rotki.icon_manager)
```

Comment:
Maybe also query icons here?

Reply:
Icons are queried periodically. So if one is deleted, the periodic task attempts to download it again. 
https://github.com/rotki/rotki/blob/develop/rotkehlchen/rotkehlchen.py#L361